x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/engines_logic.test.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { LogicMounter } from '../../../__mocks__/kea.mock';
+
+jest.mock('../../../shared/http', () => ({
+  HttpLogic: { values: { http: { get: jest.fn() } } },
+}));
+import { HttpLogic } from '../../../shared/http';
+
+import { EngineDetails } from '../engine/types';
+import { EnginesLogic } from './';
+
+describe('EnginesLogic', () => {
+  const DEFAULT_VALUES = {
+    dataLoading: true,
+    engines: [],
+    enginesTotal: 0,
+    enginesPage: 1,
+    metaEngines: [],
+    metaEnginesTotal: 0,
+    metaEnginesPage: 1,
+  };
+
+  const MOCK_ENGINE = {
+    name: 'hello-world',
+    created_at: 'Fri, 1 Jan 1970 12:00:00 +0000',
+    document_count: 50,
+    field_count: 10,
+  } as EngineDetails;
+  const MOCK_ENGINES_API_RESPONSE = {
+    results: [MOCK_ENGINE],
+    meta: {
+      page: {
+        current: 1,
+        total_pages: 10,
+        total_results: 100,
+        size: 10,
+      },
+    },
+  };
+
+  const { mount } = new LogicMounter(EnginesLogic);
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('has expected default values', () => {
+    mount();
+    expect(EnginesLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('actions', () => {
+    describe('onEnginesFetch', () => {
+      describe('dataLoading', () => {
+        it('should be set to false', () => {
+          mount();
+          EnginesLogic.actions.onEnginesFetch({ engines: [], total: 0 });
+
+          expect(EnginesLogic.values).toEqual({
+            ...DEFAULT_VALUES,
+            dataLoading: false,
+          });
```

Comment:
Jason also implemented this method of expecting/asserting against ALL `SomeLogic.values` (as opposed to just `SomeLogic.values.specificValues`), as a way of regression testing against reducer changes that we make and forget to write tests/assertions for. It's been working really well just IMO and has helped us catch bugs.